src/components/Navigation/MainNavigation/MainNavigation.component.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+import MainHeader from '../MainHeader';
+import SideDrawer from '../SideDrawer';
+import NavLinks from '../NavLinks';
+import Backdrop from '../../UIElements/Backdrop';
+import SearchInput from '../../SearchInput';
+import Toggle from '../../UIElements/Toggle';
+
+// Styled components
+const Button = styled.button`
+  width: fit-content;
+  height: 1rem;
+  background: transparent;
+  border: none;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-around;
+  cursor: pointer;
+  margin-top: 17px;
+  outline: none;
+
+  span {
+    display: block;
+    width: 1rem;
+    height: 2px;
+    background: white;
+  }
+
+  @media (min-width: 600px) {
+    margin-top: 23px;
+  }
+`;
+
+const ToggleAndLogIconContainer = styled.div`
+  display: flex;
+  align-items: center;
+  width: fit-content;
+  height: fit-content;
+  align-self: auto;
+  margin-left: auto;
+
+  .logIcon {
+    div {
+      margin: 0;
+      margin-right: 10px;
+      background-color: red;
+      width: 40px;
+      height: 40px;
+      border-radius: 31px;
+      cursor: pointer;
+    }
+  }
+`;
+
+const MainNavigation = () => {
+  const [drawerIsOpen, setDrawerIsOpen] = useState(false);
+
+  const openDrawerHandler = () => setDrawerIsOpen(true);
+
+  const closeDrawerHandler = () => setDrawerIsOpen(false);
+
+  return (
+    <>
+      {drawerIsOpen && <Backdrop onClick={closeDrawerHandler} />}
+      <SideDrawer show={drawerIsOpen} onClick={closeDrawerHandler}>
+        <nav style={{ height: '100%' }}>
+          <NavLinks />
+        </nav>
+      </SideDrawer>
+      <MainHeader>
+        <div style={{ display: 'flex', alignSelf: 'baseline' }}>
```

Comment:
I recommend not using the `style` property directly even if it is only one rule, it is better to use styled-components. We don't know if we are going to add more CSS rules on these components, having consistency on the code helps us to code faster because we don't think of the implementation, we know exactly where to put which logic or styles.